backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -14,6 +28,12 @@
   private final long endTimeAvailable;
   /** How long user wants to chat */
   private final int duration;
+  /** Role at Google */
+  private final String role;
+  /** Product area at Google */
+  private final String productArea;
```

Comment:
These fields seem more appropriate for the User object than Participant. Could they move to User instead? FindMatchQuery could load the corresponding User models for participants as well.

Reply:
Not all participants are ever added to the user datastore. They're only added to the datastore if the person opts to save preferences. And these fields are needed as part of a Participant if not all Participants have a corresponding user, unless we add every participant/user to user datastore and delete, but that would also require us to add a savePreference boolean/enum field to a Participant.

Reply:
+1 to Kevin's comment. I think the reason we use two different classes User and Participant is that `User` is used to describe a person, and `Participant` is used to indicate this person wants to participate a match. So we should only need to save the field to either one. As for opt not to save the preference, does our code support not adding `User` to datastore? If so, we can just create an object to save the input properties, but never add it to datastore.